pythonbits/scene.py

```diff
@@ -1,81 +1,383 @@
+    log.debug('Base path for {}: {}', path, dirname)
+
+    guess = _guessit(path)
+    if 'release_group' not in guess:
+        log.debug('No release group found - cannot find scene release: {}', guess)
+        return None
+
+    try:
+        if _isdir(path):
+            try:
+                _check_dir(path)
+            except SceneError as e:
+                errors_found = True
+                something_found = True
+                if on_error is not None:
+                    on_error(dirname, e)
+                else:
+                    raise
+
+        for info in _yield_file_info(path):
+            log.debug('Scene info: {}', info)
+            # Allow mixed scene/non-scene releases by only checking known scene releases
+            if info['size'] and info['crc']:
+                something_found = True
+                relpath = os.path.join(dirname, info['filename'])
+                if dirname:
+                    fspath = os.path.join(path, info['filename'])
+                else:
+                    fspath = path
+                log.debug('Torrent path: {}', relpath)
+                log.debug('File system path: {}', fspath)
+                try:
+                    _check_file(relpath, fspath, info)
+                except SceneError as e:
+                    errors_found = True
+                    if on_error is not None:
+                        on_error(relpath, e)
+                    else:
+                        raise
+    except APIError as e:
+        # Connection failed; we can't tell
+        log.debug('Ignoring error: {}', e)
+        if on_error is not None:
+            on_error(dirname or os.path.basename(path), e)
+        else:
+            raise
+    else:
+        log.debug('Something found: {}', something_found)
+        log.debug('Errors found: {}', errors_found)
+        if something_found:
+            return errors_found
+
+
+def _check_dir(path):
+    """
+    Return True if `path` was not modified or None if not sure
+
+    Raise SceneError if `path` does not match scene release
+    """
+    try:
+        release_name, files = get_details(path)
+    except APIError:
+        return None
+    log.debug('Checking directory: {}', path)
+    log.debug('release_name: {}', release_name)
+    log.debug('files: {}', files)
+    if files:
+        basename = _get_basename_noext(path)
+        log.debug('Base name should be {}: {}', release_name, basename)
+        if release_name != basename:
+            raise SceneError('%s was renamed to %s' % (release_name, basename))
+        else:
+            return True
+
+
+def _check_file(relpath, fspath, info):
+    """
+    Return True if `path` is unmodified or None if not sure
+
+    relpath: Relative path that starts with the release name
+    fspath: Path that exists in the file system
+
+    It is important that `path` is relative and starts with the release/torrent name.
+
+    Raise SceneError if `path` does not match `info`
+    """
```

Comment:
call signature does not match documentation

Reply:
Fixed.